### PR TITLE
Wrapping errors when cleaning up pods

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Improve debuggability of Pod cleanup

#### Which issue(s) this PR fixes:

Part of #1591

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by tenzen-y** [comment]
#1591 happened.

**Comment by alculquicondor** [comment]
Also #1593

File: test/util/util.go [hunk 1]
```diff
@@ -137,13 +137,13 @@ func DeleteAllJobsInNamespace(ctx context.Context, c client.Client, ns *corev1.N
 func DeleteAllPodsInNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace) error {
 	err := c.DeleteAllOf(ctx, &corev1.Pod{}, client.InNamespace(ns.Name))
 	if err != nil && !apierrors.IsNotFound(err) {
-		return err
+		return fmt.Errorf("deleting all Pods in namespace %v: %w", ns.Name, err)
```
Comment (tenzen-y) [inline test/util/util.go:140]:
```suggestion
		return fmt.Errorf("deleting all Pods in namespace %q: %w", ns.Name, err)
```

Comment (tenzen-y) [reply test/util/util.go:140]: Or

```suggestion
		return fmt.Errorf("deleting all Pods in namespace %s: %w", ns.Name, err)
```

File: test/util/util.go [hunk 2]
```diff
@@ -137,13 +137,13 @@ func DeleteAllJobsInNamespace(ctx context.Context, c client.Client, ns *corev1.N
...
 	}
 
 	lst := corev1.PodList{}
 	err = c.List(ctx, &lst, client.InNamespace(ns.Name))
 	if err != nil && !apierrors.IsNotFound(err) {
-		return err
+		return fmt.Errorf("listing Pods with a finalizer in namespace %v: %w", ns.Name, err)
```
Comment (tenzen-y) [inline test/util/util.go:146]: ditto.

**Review by tenzen-y — COMMENTED** [review]
Basically lgtm
Just some nits.

File: test/util/util.go (hunk 1 above)
Comment (alculquicondor) [reply test/util/util.go]: Done as `%q`

File: test/util/util.go (hunk 2 above)
Comment (alculquicondor) [reply test/util/util.go]: Done

File: test/e2e/singlecluster/pod_test.go [hunk 3]
```diff
@@ -168,7 +168,7 @@ var _ = ginkgo.Describe("Pod groups", func() {
 			})
 		})
 
-		ginkgo.It("Failed Pod can be replaced in group", func() {
+		ginkgo.FIt("Failed Pod can be replaced in group", func() {
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply test/e2e/singlecluster/pod_test.go]: :( fixed

**Review by tenzen-y — COMMENTED** [review]
Thanks!
/lgtm
/approve

**Pull request merged**